tests/integration/cli/tests/run2.rs

```diff
@@ -0,0 +1,339 @@
+    use super::*;
+    use predicates::str::contains;
+
+    #[test]
+    fn wasi_executable() {
+        let assert = Command::new(get_wasmer_path())
+            .arg("run2")
+            .arg(fixtures::qjs())
+            .arg("--")
+            .arg("--eval")
+            .arg("console.log('Hello, World!')")
+            .assert();
+
+        assert.success().stdout(contains("Hello, World!"));
+    }
+
+    #[test]
+    fn no_abi() {
+        let assert = Command::new(get_wasmer_path())
+            .arg("run2")
+            .arg(fixtures::fib())
+            .assert();
+
+        assert.success();
+    }
+
+    #[test]
+    fn error_if_no_start_function_found() {
+        let assert = Command::new(get_wasmer_path())
+            .arg("run2")
+            .arg(fixtures::wat_no_start())
+            .assert();
+
+        assert
+            .failure()
+            .stderr(contains("The module doesn't contain a \"_start\" function"));
+    }
+
+    #[test]
+    fn pre_compiled() {
+        let temp = TempDir::new().unwrap();
+        let dest = temp.path().join("qjs.wasmu");
+        let qjs = fixtures::qjs();
+        // Make sure it is compiled
+        Command::new(get_wasmer_path())
+            .arg("compile")
+            .arg("-o")
+            .arg(&dest)
+            .arg(&qjs)
+            .assert()
+            .success();
+        assert!(dest.exists());
+
+        // Now we can try to run the compiled artifact
+        let assert = Command::new(get_wasmer_path())
+            .arg("run2")
+            .arg(&dest)
+            .arg("--")
+            .arg("--eval")
+            .arg("console.log('Hello, World!')")
+            .assert();
+
+        assert.success().stdout(contains("Hello, World!"));
+    }
+}
+
+#[test]
+fn wasmer_package_directory() {
+    let temp = TempDir::new().unwrap();
+    std::fs::copy(fixtures::qjs(), temp.path().join("qjs.wasm")).unwrap();
+    std::fs::copy(fixtures::qjs_wasmer_toml(), temp.path().join("wasmer.toml")).unwrap();
+
+    let assert = Command::new(get_wasmer_path())
+        .arg("run2")
+        .arg(temp.path())
+        .arg("--")
+        .arg("--eval")
+        .arg("console.log('Hello, World!')")
+        .assert();
+
+    assert.success().stdout(contains("Hello, World!"));
+}
+
+mod remote_webc {
+    use super::*;
+
+    #[test]
+    fn quickjs_as_package_name() {
+        let assert = Command::new(get_wasmer_path())
+            .arg("run2")
+            .arg("saghul/quickjs")
+            .arg("--entrypoint=quickjs")
+            .arg("--registry=https://wapm.io/")
+            .arg("--")
+            .arg("--eval")
+            .arg("console.log('Hello, World!')")
+            .assert();
+
+        assert.success().stdout(contains("Hello, World!"));
+    }
+
+    #[test]
+    fn quickjs_as_url() {
+        let assert = Command::new(get_wasmer_path())
+            .arg("run2")
+            .arg("https://wapm.io/saghul/quickjs")
+            .arg("--entrypoint=quickjs")
```

Comment:
What happens if we don't provide an entrypoint (and there is only one command / or the package has already an entrypoint defined)? Will things still work?

Reply:
Yep!

https://github.com/wasmerio/wasmer/blob/bca161cb441b769722b7d7ca1ddf4d30aa5332e8/lib/cli/src/commands/run2.rs#L271-L287

I've also added an extra test using WABT to show the message you get when you need to specify the entrypoint.

https://github.com/wasmerio/wasmer/blob/bca161cb441b769722b7d7ca1ddf4d30aa5332e8/tests/integration/cli/tests/run2.rs#L51-L60